server/container/core/src/test/java/org/apache/james/server/core/MailImplTest.java

```diff
@@ -292,4 +301,68 @@ void mailImplShouldNotAllowSettingEmptyName() {
         assertThatThrownBy(() -> newMail().setName(""))
             .isInstanceOf(IllegalArgumentException.class);
     }
+
+    @Test
+    void mailImplShouldBeSerializable() throws Exception {
+        MailImpl mail = MailImpl.builder()
+            .name("mail-id")
+            .sender(MailAddress.nullSender())
+            .build();
+
+        ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+        ObjectOutputStream objectOutputStream = new ObjectOutputStream(byteArrayOutputStream);
+        objectOutputStream.writeObject(mail);
+        objectOutputStream.close();
+        ByteArrayInputStream byteArrayInputStream = new ByteArrayInputStream(byteArrayOutputStream.toByteArray());
+        ObjectInputStream objectInputStream = new ObjectInputStream(byteArrayInputStream);
+        Object object = objectInputStream.readObject();
+
+        assertThat(object)
+            .isInstanceOf(MailImpl.class)
+            .isEqualTo(mail);
```

Comment:
if you introduced equality for that purpose, you can do a compare field by field ignoring `message` field instead.

Reply:
Yes I did ^^